scheduler/scheduler.go

```diff
@@ -127,8 +128,47 @@ func (s *Scheduler) processExecution(exec *execution.Execution) {
 			s.ErrC <- err
 			continue
 		}
-		if _, err := s.execution.Execute(wf.Hash, nextStep.InstanceHash, nil, exec.Hash, nextStepID, nextStep.TaskKey, exec.Outputs, []string{}); err != nil {
+		edge, err := wf.FindEdge(exec.StepID, nextStepID)
```

Comment:
The functions `ChildrenIDs` and `FindEdge` should be replaced by `EdgesFrom(src nodeKey)`. Both function are doing almost the same thing.

Reply:
Same apply to `ParentIDs`. Returning the edges instead of the just the node key is more reusable.